### PR TITLE
applications: nrf5340_audio: Update changelog

### PR DESCRIPTION
**Pull request by erikrobstad**
For the NCS 2.5.0 release.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -274,12 +274,15 @@ nRF5340 Audio
...
   * Management - This module handles scanning and advertising, in addition to general initialization, controller configuration, and transfer of DFU images.
+    The new architecture makes it possible to make connections and handle periodic advertising sync independently of the Bluetooth LE Audio setup.
   * Stream - This module handles the setup and transfer of audio in the Bluetooth LE Audio context.
+    The new architecture makes it possible to have more than one Bluetooth LE Audio role in one device.
   * Renderer - This module handles rendering, such as volume up and down.
   * Content Control - This module handles content control, such as play and pause.
 
 * Added back the QDID number for the :ref:`lib_bt_ll_acs_nrf53_readme` to the documentation.
 * Updated the :ref:`application documentation <nrf53_audio_app>` by splitting it into several pages.
+* Fixed a compatibility issue when using Android 14.
```
Comment (rick1082) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:285]: Not sure if we should add "Pixel 7/7A".

**Review by rick1082 — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -274,12 +274,15 @@ nRF5340 Audio
 * The whole application architecture for handling Bluetooth LE Audio has changed. New modules have been added:
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:274]:
```suggestion
* Modified the entire application architecture for handling Bluetooth LE Audio. The following new modules have been added:
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (erikrobstad) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:285]: Changed to "Google Pixel 7/7A"

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (erikrobstad) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Done

**Review by greg-fer — APPROVED** [review]

**Pull request merged**